pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -113,30 +113,28 @@ def attach_dataloaders(
     def attach_datamodule(
         self, model: 'pl.LightningModule', datamodule: Optional['pl.LightningDataModule'] = None
     ) -> None:
-        # We use datamodule if it's been provided, otherwise we check model for it
```

Comment:
unrelated changes?

Reply:
`model.datamodule` is deprecated so I removed the deprecated usage otherwise it gets printed everywhere

Reply:
so you removed `datamodule = datamodule or getattr(model, 'datamodule', None)`
and pulled out `if datamodule`

Reply:
Exactly